plugin/db/mysql/mysql.go

```diff
@@ -938,6 +981,78 @@ func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner) (err error
 	return nil
 }
 
+// Ghost table names are generated by appending "_ghost" to the original ones.
+// Constraint names are generated by appending a timestamp (to second) to the original ones.
+// According to https://dev.mysql.com/doc/refman/8.0/en/identifier-length.html, the max length of a
+// constraint name is 64, so we should truncate the original constraint name part to fit in 64 characters.
+func rewriteToGhostTableForRestore(stmt, timestamp string) (string, error) {
+
+	// matchesInsert would be like:
+	// with dbPrefix: []string{"INSERT INTO `dbPrefix`.`table` VALUES (1, 2, 3);", "dbPrefix", "table", "(1, 2, 3);"}
+	// empty dbPrefix: []string{"INSERT INTO `table` VALUES (1, 2, 3);", "", "table", "(1, 2, 3);"}
+	matchesInsert := reInsert.FindStringSubmatch(stmt)
+
+	// matchesCreate would be like:
+	// []string{"CREATE TABLE `table` (", "table", "("}
+	matchesCreate := reCreate.FindStringSubmatch(stmt)
+
+	if matchesInsert != nil {
+		dbPrefix := matchesInsert[1]
+		if dbPrefix != "" {
+			// Currently we only consider restoring a single database, if db prefix is matched, it is unexpected
+			return "", fmt.Errorf("expect dbPrefix to be empty, but get %q", dbPrefix)
+		}
+		ghostTable := fmt.Sprintf("%s_ghost", matchesInsert[2])
+		values := matchesInsert[3]
+		return fmt.Sprintf("INSERT INTO %s`%s` VALUES %s", dbPrefix, ghostTable, values), nil
```

Comment:
We should not have dbPrefix anymore.

Reply:
BTW, even we keep it, seems we miss the "." (dot) ?

Reply:
According to a former discussion at https://github.com/bytebase/bytebase/pull/1050#discussion_r850089808, we agreed to regard a dbPrefix here as unexpected. And I should remove it from the rewritten SQL (which I forgot, sorry :/)